src/main/java/scmspain/karyon/restrouter/transport/http/RouteInterceptorSupport.java

```diff
@@ -0,0 +1,74 @@
+package scmspain.karyon.restrouter.transport.http;
+
+import io.netty.buffer.ByteBuf;
+import io.reactivex.netty.protocol.http.server.HttpServerRequest;
+import io.reactivex.netty.protocol.http.server.HttpServerResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * Created by borja.vazquez on 25/9/15.
+ */
+public class RouteInterceptorSupport {
+  private static final Logger logger = LoggerFactory.getLogger(RouteInterceptorSupport.class);
+
+  private List<RouteOutInterceptor> routeOutInterceptorList = new ArrayList<>();
+
+  public RouteInterceptorSupport() {
+    routeOutInterceptorList.add((result, request, response) -> result.flatMap(value -> {
+      if (value == null) {
+        return Observable.empty();
+      } else {
+        String message = "Cannot handle value of type as Observable type in endpoint return: " + value.getClass();
+        logger.error(message);
+        return Observable.error(new RuntimeException(message));
+      }
+    }));
+  }
+
+  public void addOutInterceptor(RouteOutInterceptor routeOutInterceptor) {
+    routeOutInterceptorList.add(routeOutInterceptorList.size()-1, routeOutInterceptor);
+  }
+
+  /**
+   * It executes all interceptor in order to the response body.
+   * Return generic type will be Void for sure, because we add an interceptor at the end that will
+   * convert the value to a Void.
+   * @param responseBodyObs the responseBody Observable
+   * @param request the request
+   * @param response the response
+   * @return the Void observable that karyon expects
+   */
+  @SuppressWarnings("unchecked")
+  public Observable<Void> execute(Observable<?> responseBodyObs, HttpServerRequest<ByteBuf> request, HttpServerResponse<ByteBuf> response) {
+    return (Observable<Void>)applyInterceptors(responseBodyObs, request, response, 0);
```

Comment:
I don't think using an Observable just to have a more convenient way to reduce/fold the interceptors makes this better. Java8 stream API should have this kind of operation, but due to the way they do the parallel streams reduce has to be this way. (unfortunately this is not Scala where just using a foldLeft this is done in a cleaner way).

If the last interceptor intercepts an Observable of String to write in the socket you are forcing to write only text data not binary and you are losing the ability to write in a custom charset.

Another approach is to write byte[] at the last observable but imho is quite weird, and is less flexible as you are forced to work with byte[].